pkg/pgctl/pgrestore.go

```diff
@@ -31,16 +34,44 @@ func (x *Restore) Exec(filename string, opts ExecOptions) Result {
 	result.FullCommand = strings.Join(options, " ")
 	cmd := exec.Command(PSQL, options...)
 
-	//cmd.Env = append(os.Environ(), x.EnvPassword)
+	// Pipe to capture error output
 	stderrIn, _ := cmd.StderrPipe()
 	go func() {
 		result.Output = streamExecOutput(stderrIn, opts)
 	}()
-	cmd.Start()
-	err := cmd.Wait()
-	if exitError, ok := err.(*exec.ExitError); ok {
-		result.Error = &ResultError{Err: err, ExitCode: exitError.ExitCode(), CmdOutput: result.Output}
+
+	err := cmd.Start()
+	if err != nil {
+		result.Error = &ResultError{Err: err, CmdOutput: result.Output}
+		return result
+	}
+
+	err = cmd.Wait()
+	if err != nil {
+		var exitError *exec.ExitError
+		if errors.As(err, &exitError) {
```

Comment:
I think you meant this. This is a pointer to a pointer
```suggestion
		if errors.As(err, exitError) {
```

Reply:
Fixed.